lambdas/account-scoped/src/conversation/deleteClosedConversationHandler.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (C) 2021-2025 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+import {
+  registerServiceScopedConversationEventHandler,
+  ServiceScopedConversationEventHandler,
+} from './serviceScopedConversationEventHandler';
+import type { AccountSID } from '@tech-matters/twilio-types';
+import { Twilio } from 'twilio';
+import {
+  CHANNEL_UPDATED,
+  ChannelUpdatedEvent,
+  ConversationStateUpdatedEvent,
+} from './eventTypes';
+import { retrieveServiceConfigurationAttributes } from '../configuration/aseloConfiguration';
+import { getChatServiceSid } from '@tech-matters/twilio-configuration';
+
+const deleteClosedConversationEventHandler: ServiceScopedConversationEventHandler =
+  async (
+    {
+      StateTo: stateTo,
+      StateFrom: stateFrom,
+      ConversationSid: conversationSid,
+    }: ConversationStateUpdatedEvent,
+    accountSid: AccountSID,
+    client: Twilio,
+  ): Promise<void> => {
+    const serviceConfigAttributes = await retrieveServiceConfigurationAttributes(client);
+
+    const zeroTranscriptRetention = Boolean(
+      serviceConfigAttributes.enforceZeroTranscriptRetention || 'false',
+    );
+    if (zeroTranscriptRetention && stateTo === 'closed') {
+      console.debug(
+        `Conversation state for ${conversationSid} updated from ${stateFrom} to ${stateTo} and account ${accountSid} has a zero transcript retention policy, deleting conversation`,
+      );
+      await client.conversations.v1.conversations.get(conversationSid).remove();
```

Comment:
```suggestion
      await client.conversations.v1.conversations.get(conversationSid).remove({ xTwilioWebhookEnabled: 'true' });
```
needs to be set if we want this action to raise events I believe. Otherwise Twilio won't call the webhooks for the event. https://www.twilio.com/docs/conversations/conversations-webhooks#triggering-webhooks-for-rest-api-events

We may don't care about raising this event just mentioning in case we do.